deploy-board/deploy_board/templates/host_side_panel.html

```diff
@@ -42,6 +46,39 @@ <h4 class="modal-title">Host Termination Confirm</h4>
         </div><!-- /.modal-content -->
     </div><!-- /.modal-dialog -->
 </div><!-- /.modal -->
+
+<script>
+    $("#terminateBtn").click(function () {
+        async function doAjax(args) {
+            let result;
+            try {
+                result = await $.ajax({
+                    type: 'GET',
+                    url: `/groups/{{env_name}}-{{stage_name}}/hosts`,
+                    dataType: "json",
+                    beforeSend: function(xhr, settings) {
+                        var csrftoken = getCookie('csrftoken');
+                        xhr.setRequestHeader("X-CSRFToken", csrftoken);
+                    },
+                });
+                return result.length;
+            } catch (error) {
+                console.error(error);
+                return 0;
+            }
+        };
+        doAjax().then( (data) => {
```

Comment:
You don't have to call backend at all if `{{termination_limit}} === null`.
Indeed you don't even have to render the UI if `termination_limit` is not set.